tsl/src/remote/txn.c

```diff
@@ -278,29 +279,36 @@ remote_txn_check_for_leaked_prepared_statements(RemoteTxn *entry)
 {
 	PGresult *res;
 	char *count_string;
+	ExecStatusType status;
 
 	if (PQTRANS_IDLE != PQtransactionStatus(remote_connection_get_pg_conn(entry->conn)))
 		return;
 
 	res = remote_connection_exec(entry->conn, "SELECT count(*) FROM pg_prepared_statements");
```

Comment:
I guess this will fail in case if prepared statements are used in some other non-related cases, we likely need to filter then in order to associate with the main transaction